utils/p4runtime_lib/convert.py

```diff
@@ -42,6 +42,23 @@ def encodeIPv4(ip_addr_string):
 
 def decodeIPv4(encoded_ip_addr):
     return socket.inet_ntoa(encoded_ip_addr)
+    
+ipv6_pattern = re.compile(r'([a-f0-9]+:+)+[a-f0-9]+')
```

Comment:
I would recommend removing the regex pattern matching for IPv6 addresses completely.  This regex matches things that are _not_ legal IPv6 addresses.

Reply:
And it fails to match things that _are_ legal IPv6 addresses, which is the more important point for the way it is used elsewhere.

Reply:
Thanks for the input! I've removed the regex pattern.